Grilled Cheese Sandwich Entity and Recipe

AgentSmithRadio opened the pull request:
## About the PR

[NEW VIDEO!!!](https://www.youtube.com/watch?v=XNzCVV6uoAg)

I made a Grilled Cheese Sandwich entity. It is made with two slices of bread, a cheese wedge and a slice of butter, cooked for 10 seconds in the microwave.

## Why / Balance
Grilled cheese is a beloved food with pre-existing sprites from SS13. It's a natural companion food for the soups that chefs can make, and it creates a practical way to use bread slices without having to rely on botany. Bread slices don't have many recipes as it currently stands.

## Technical details
Added a grilled cheese sprite used commonly in SS13, made an edit to its relevant meta.json file so entities can use it, added the recipe to meal_recipes.yml, added the entity to misc.yml, and added the food as a random spawn with food_baked_single.yml. 

## Media

[NEW VIDEO!!!](https://www.youtube.com/watch?v=XNzCVV6uoAg)

## Requirements
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.


## Breaking changes
This PR is entirely additive, it shouldn't cause any issues.

**Changelog**
:cl: AgentSmithRadio
- add: Added grilled cheese sandwich and recipe.

AgentSmithRadio commented:
Before this can be approved, I need help with the CC attribution for meta.json. I can provide details on where I got my sprite to anyone who is an expert in sourcing these images.

AgentSmithRadio commented:
I've been struggling with attribution for the grilled cheese sprite. Currently I have "Grilled-cheese taken from ParadiseSS13 at commit https://github.com/ParadiseSS13/Paradise/commit/e8ab249edacb70373d80bd4308e74fe7dd532fbd", and I can verify that the sprite image is [CC-BY-SA 3.0 license](https://creativecommons.org/licenses/by-sa/3.0/) via https://paradisestation.org/wiki/index.php?title=File:Grilled_Cheese.png. If I have this wrong please let me know, but sourcing the image any older than this point hasn't yielded any results.

AgentSmithRadio commented:
https://github.com/space-wizards/space-station-14/pull/33508 caused some pretty severe merge conflicts. I rebased, tested grilled cheese, the new entities, and the muffin food entities that were caught in the merge conflict crossfire, and force-pushed as a result of the rebase.

beck-thompson commented on Resources/Prototypes/Entities/Objects/Consumable/Food/Baked/misc.yml:
```diff
@@ -835,3 +835,28 @@
+      food:
+        maxVol: 19
+        reagents:
+        - ReagentId: Nutriment
```
Just make these be exactly what the input ingredients add up to. I'm pretty sure it should be:
Nutriment: 4.00 + 4.00 + 3.75
Butter: 5.00

Changing some of the nutriment to vitamin is also fine just make sure the totals add up

beck-thompson commented on Resources/Textures/Objects/Consumable/Food/Baked/misc.rsi/meta.json:
```diff
@@ -1,7 +1,7 @@
     "version": 1,
     "license": "CC-BY-SA-3.0",
-    "copyright": "Taken from tgstation and modified by Swept at https://github.com/tgstation/tgstation/commit/40d75cc340c63582fb66ce15bf75a36115f6bdaa. Chevrechaud created by Github user deathride58, croissant taken from tgstation at commit https://github.com/tgstation/tgstation/commit/7ffd61b6fa6a6183daa8900f9a490f46f7a81955. Muffin-chocolate, muffin-banana and muffin-cherry sprites modified fro ...
+    "copyright": "Taken from tgstation and modified by Swept at https://github.com/tgstation/tgstation/commit/40d75cc340c63582fb66ce15bf75a36115f6bdaa. Chevrechaud created by Github user deathride58, croissant taken from tgstation at commit https://github.com/tgstation/tgstation/commit/7ffd61b6fa6a6183daa8900f9a490f46f7a81955. Muffin-chocolate, muffin-banana and muffin-cherry sprites modified fro ...
```
You should use the original commit! Looks like this was just from rearranging. Take a look through the git history and figure out which one it was added in!

https://github.com/Qwertytoforty/Paradise/commits/ce433d9755f98febe521a06e677490af0be62968/icons/obj/food/food.dmi

beck-thompson reviewed (changes requested):
Could you move the sprite up some amount? It being so low down is odd!

![image](https://github.com/user-attachments/assets/36189972-085d-4dbe-b3b8-c58fcf27dab6)

AgentSmithRadio commented:
A few changes. I changed the cook time to 10 seconds (more in line with a taco, 5 feels too short for "toasting" something), fixed the sprite to be properly centered, found the actual original creator/commit for the sprite and changed my attribution accordingly, and changed the nutritional information to the following:

21 Capacity - 5u free
11 Nutriment
2 Butter
3 Vitamin

In short, I rounded down the .75 from the cheese, and used butter to convert into vitamin instead of nutriment. The butter is primarily there to match other foods with butter having it as a reagent, but also to impart the butter flavour descriptor to the food. At 16 units of nutrition this sandwich is VERY strong for the ingredients required, the low-end burgers are worse than this.

beck-thompson reviewed (approved):
Looks great!

UbaserB reviewed (approved):
art approval
